### PR TITLE
Allow disabling experimental commands, devices on stable branch

### LINKED ISSUE #30152: Add configuration to disable certain tool features on stable branches
Allow us to clearly message which features are stable or not.

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

To make it clear which features are still experimental, these should be disabled for users on "stable" branches.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/30152

## Tests

I added the following tests:

- flutter_command_test.dart: Experimental command can be disabled on stable branch.
- flutter_command_test.dart: Experimental command Works normally on regular branches

These verify that executing an experimental command on a stable branch throws a tool exit.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
Thank you @jonahwilliams.  FYI @timsneath @kevmoo

File: packages/flutter_tools/lib/src/desktop.dart [hunk 1]
```diff
@@ -3,11 +3,12 @@
...
 
 import 'base/platform.dart';
+import 'version.dart';
 
 // Only launch or display desktop embedding devices if
 // `FLUTTER_DESKTOP_EMBEDDING` environment variable is set to true.
 bool get flutterDesktopEnabled {
   _flutterDesktopEnabled ??= platform.environment['FLUTTER_DESKTOP_EMBEDDING']?.toLowerCase() == 'true';
-  return _flutterDesktopEnabled;
+  return _flutterDesktopEnabled && FlutterVersion.instance.getBranchName() != 'stable';
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/desktop.dart]: Can 'stable' be a constant or an enum member? Or maybe instead:

`FlutterVersion.instance.isStable`

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/desktop.dart]: Done

**Review by zanderso — APPROVED** [review]
lgtm

**Pull request merged**